From 1d326505482a34a023493576398c29761e92eb85 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 11 May 2018 17:38:29 +0800
Subject: [PATCH 3791/5242] MLK-18206-5 usb: cdns3: gadget: change maxburst as
 numbered buffered packet

commit  535e9678397bf5af7d8e9846a8a511890e397792 from
https://source.codeaurora.org/external/imx/linux-imx.git

This value is suggested by CDNS IC engineer.

Reviewed-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index 447f8b7..c7a3e14 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -2217,7 +2217,7 @@ static int usb_ss_init_ep(struct usb_ss_dev *usb_ss)
 		if (bulk_ep_reg & (1uL << ep_reg_pos)) {
 			usb_ss_ep->endpoint.caps.type_bulk = 1;
 			usb_ss_ep->endpoint.caps.type_int = 1;
-			usb_ss_ep->endpoint.maxburst = 15;
+			usb_ss_ep->endpoint.maxburst = CDNS3_EP_BUF_SIZE - 1;
 		}
 
 		list_add_tail(&usb_ss_ep->endpoint.ep_list,
-- 
1.7.9.5

